### PR TITLE
Dev containerservice microsoft.container service 2020 04 01

### PR DESCRIPTION
**Pull request by gtxistxgao**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding) for the workflow.
- [ ] [Validation tools](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/84/Swagger-Validation-tools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/FAQ.md).

### ARM API Review Checklist
- [ ] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process]( https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review).

### CODE REVIEW

**Comment by gtxistxgao** [comment]
Create this PR to get the temp SDK. 2020-04-01 has not yet published. Please don't merge.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json [hunk 1]
```diff
@@ -1997,7 +1997,7 @@
           "type": "object",
           "properties": {
             "balance-similar-node-groups": {
-              "type": "boolean"
+              "type": "string"
```
Comment (marwanad) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:2000]: This change is already in master:

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json#L1999

This change is already in master. I think the dev branch might need rebasing but we don't have access to that.

Comment (marwanad) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:2000]: cc @fengzhou-msft/anyone who has access

**Comment by gtxistxgao** [comment]
@fengzhou-msft  ready to be merged

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json [hunk 2]
```diff
@@ -1997,7 +1997,7 @@
           "type": "object",
           "properties": {
             "balance-similar-node-groups": {
-              "type": "boolean"
```
Comment (ravbhatnagar) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:2000]: please confirm this (whether it a issue which will be fixed with rebasing) else this change is a breaking change and would have to go in a  new api-version

**Comment by ravbhatnagar** [comment]
Based on comment from @marwanad it may be a rebasing issue. If that is the case, the consider this signed off from ARM side. @fengzhou-msft  - FYI and you can add the ARMSIgnedOff label if its a rebasing issue.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json (hunk 2 above)
Comment (marwanad) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:2000]: yup, that's a rebase issue:

The master branch is here: 

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json#L2000

**Review by KrisBash — APPROVED** [review]
Per Gaurav, this should be signed off

**Pull request merged**